emulators/mame/Portfile

```diff
@@ -14,52 +14,69 @@ PortSystem          1.0
+    hardware and how it functions. The source code to MAME serves as this \
     documentation.
-
-github.tarball_from archive
+homepage            https://www.mamedev.org
 
 #------------------------------------------------------------------------------
-# Patches and Checksums
+# Versions, Patches, and Checksums
 #------------------------------------------------------------------------------
 
-# Patch 'language-portuguese_brazil' applies to release 0.226; fixes
-# errors during translation build phase.
-# Fix submitted, and merged, by Mame maintainers; will be part of Mame
-# releases from 0.227 onward. Tracked by Mame issue:
-# https://github.com/mamedev/mame/issues/7510
-#
-# Patch 'src-posixfile-build-error' applies to release 0.226; fixes
-# compilation errors for MacOS 10.8 and Xcode 5.x.
-# Fix submitted, and merged, by Mame maintainers; will be part of Mame
-# releases from 0.227 onward. Tracked by Mame issue:
-# https://github.com/mamedev/mame/issues/7536
-#
-# Patch 'scripts-genie-compile-warnings' applies to all releases; greatly
-# reduces compilation warnings.
-
-patchfiles          mame-patch-0226-language-portuguese_brazil.diff \
-                    mame-patch-0226-src-posixfile-build-error.diff \
-                    mame-patch-0226-scripts-genie-compile-warnings.diff
-
-checksums           rmd160  b9e4ae321b7673790d374c63bbe966d1502d6738 \
+if {${name} eq ${subport}} {
+    revision        3
+    conflicts       ${name}-latest
+
+    set g_mame_release \
+                    "0226"
+
+    # Patch 'language-portuguese_brazil' applies to release 0.226; fixes
+    # errors during translation build phase.
+    # Merged into Mame releases from 0.227 onward. Tracked by issue:
+    # https://github.com/mamedev/mame/issues/7510
+    #
+    # Patch 'src-posixfile-build-error' applies to release 0.226; fixes
+    # compilation errors for MacOS 10.8 and Xcode 5.x.
+    # Merged into Mame releases from 0.227 onward. Tracked by issue:
+    # https://github.com/mamedev/mame/issues/7536
+    patchfiles      \
+                    mame-patch-0226-language-portuguese_brazil.diff \
+                    mame-patch-0226-src-posixfile-build-error.diff
+
+    checksums       \
+                    rmd160  b9e4ae321b7673790d374c63bbe966d1502d6738 \
                     sha256  7c4c9ec232ba988e65fd29665c9b8e40b5ac3aa9f561eeb107cebbf08ba94baf \
                     size    196379874
+}
+
+subport ${name}-latest {
+    revision        0
+    conflicts       ${name}
+
+    set g_mame_release \
+                    "0227"
+
+    checksums       \
+                    rmd160  862df6a449329fe5b16e6d46a030fe157c3a8eb3 \
+                    sha256  95dbce00a4f05a35f66ef966fe9efad1e4e78ce62e0eba3f7031dfa6737829a5 \
+                    size    195226157
+}
+
+github.setup        mamedev mame ${g_mame_release} mame
+version             [string index ${github.version} 0].[string range ${github.version} 1 end]
```

Comment:
I'm not clear what this will set the version to. But we want it to be monotonically increasing. Just set the version to be the date of the commit in year-month-day format. For example:

`version   2021-01-08`

Reply:
To avoid maintaining two version strings, we take the one used for artifacts - such as `0227` - and simply insert a period between the first two characters. With the result being `0.227`, which is the true user-presentable version number. (Mame releases still have a major version of zero, after all these years. Seems odd, but hey... to each their own...)

We now have a formal variable with that string - `g_mame_release` - so we can at least reference that instead, rather than `github.version`. I'll also add a brief comment, reiterating the above explanation.

How does that sound?

Reply:
Changes pushed.